### DISCUSSION TITLE
Is it possible for python to use OrtSessionOptionsAppendExecutionProvider_DML function?

### DISCUSSION THREAD

**Discussion by JiaPai12138** [microsoft/onnxruntime#10615] [Other Q&A] [answered]
I can set it manually through graphic settings, but how to do it in program?

**Comment by skottmckay** [comment, marked as answer]
You would need a build that included DML. You can check the output from `get_available_providers` to make sure the DML execution provider is listed. 


```python
import onnxruntime as ort
# include all execution providers. alternatively you could use providers=['DmlExecutionProvider']
sess = ort.InferenceSession('model.onnx', providers=ort.get_available_providers())
``

**Reply by JiaPai12138** [reply]
> You would need a build that included DML. You can check the output from `get_available_providers` to make sure the DML execution provider is listed.
> 
> ```python
> import onnxruntime as ort
> # include all execution providers. alternatively you could use providers=['DmlExecutionProvider']
> sess = ort.InferenceSession('model.onnx', providers=ort.get_available_providers())
> ``
> ```

I know and I did. And i found by default it will run on my nvidia device. The problem is that i also need to use dxgi screenshot, which must be run on my Intel UHD graphic card. Seems like i cannot handle both at the same time

**Reply by skottmckay** [reply]
I believe DML has a device_id option. This may help:

```python
providers = [
    ('DmlExecutionProvider', {
        'device_id': 0  # default is 0 so you may need to set this to 1 to run on the other graphics card
    }),
    'CPUExecutionProvider',
]

sess = ort.InferenceSession('model.onnx', providers=providers)
```

**Reply by JiaPai12138** [reply]
Thanks very much! That works well!
